datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -516,7 +517,7 @@ fn statistical_join_selection_subrule(
 pub type PipelineFixerSubrule =
     dyn Fn(Arc<dyn ExecutionPlan>, &ConfigOptions) -> Result<Arc<dyn ExecutionPlan>>;
 
-/// Converts a hash join to a symmetric hash join in the case of infinite inputs on both sides.
+/// Converts a hash join to a symmetric hash join if infinite and incremental inputs exist on both sides.
```

Comment:
Based on the code, I think unbounded is the requirement, it can have finite or infinite mem